### ISSUE TITLE
Build Sync fails when toolchain is pinned to version 1.55

### ISSUE THREAD

**Issue by jmfrank63** [intellij-rust/intellij-rust#8260]
## Environment

* **IntelliJ Rust plugin version:** 0.4.161.4292-213 and 0.4.162.4301-213-nightly
* **Rust toolchain version:** 1.55
* **IDE name and version:** Both IntelliJ and CLion 2021.3
* **Operating system:** Ubuntu 20.04

## Problem description
My toolchain is currently pinned to 1.55 due to issues with compile times. Build: Sync now fails with:

```
Execution failed (exit code 101).
/home/johannes-enhance/.cargo/bin/cargo metadata --verbose --format-version 1 --all-features
stdout : error: failed to download `hermit-abi v0.1.20`

Caused by:
  unable to get packages from source

Caused by:
  failed to parse manifest at `<masked home>/.cargo/registry/src/github.com-1ecc6299db9ec823/hermit-abi-0.1.20/Cargo.toml`

Caused by:
  feature `edition2021` is required

  The package requires the Cargo feature called `edition2021`, but that feature is not stabilized in this version of Cargo (1.55.0 (32da73ab1 2021-08-23)).
  Consider trying a newer version of Cargo (this may require the nightly release).
  See https://doc.rust-lang.org/nightly/cargo/reference/unstable.html#edition-2021 for more information about the status of this feature.

stderr : 
```

## Steps to reproduce

Open a rust project and pin the compiler version to 1.55 with `rustup override set 1.55`

Possibly use a workspace, i.e., use different branches.

Build sync will fail.

I can make the project build sync work by `rustup override set 1.56.1` then switch back to 1.55 by `rustup override set 1.55`. But on the next sync I have to repeat the process again.

My project has no dependency on `hermit-abi`, so the compiler version of my project should not influence the behaviour of the plugin.

**Comment by vlad20012** [comment]
I think this is related to #7185

**Comment by Undin** [comment]
I suppose it's the same as #8517 that was fixed in #8609

@jmfrank63 could you verify it?
